### PR TITLE
auto pos creator v2

### PR DESCRIPTION
**Pull request by psorinionut**

### CODE REVIEW

File: auto-pos-creator/src/multi_contract_interactions/create_pos.rs [hunk 1]
```diff
@@ -1,231 +1,202 @@
...
-            output_payments.push(enter_result.new_farm_token);
-
-            return output_payments.send_and_return(&args.caller);
+    fn process_payment(
+        &self,
+        payment: EgldOrEsdtTokenPayment,
+        swap_operations: MultiValueEncoded<SwapOperationType<Self::Api>>,
+    ) -> EsdtTokenPayment {
+        let esdt_payment = self.get_esdt_payment(payment);
+        require!(esdt_payment.token_nonce == 0, "Only fungible ESDT accepted");
```
Comment (CostinCarabas) [inline auto-pos-creator/src/multi_contract_interactions/create_pos.rs:31]: `esdt_payment.token_nonce == 0` is also checked in the `get_esdt_payment` function.

File: auto-pos-creator/Cargo.toml [hunk 2]
```diff
@@ -37,6 +34,10 @@ rev = "bd02b83"
 git = "https://github.com/multiversx/mx-exchange-sc"
 rev = "bd02b83"
 
+[dependencies.router]
+git = "https://github.com/multiversx/mx-exchange-sc"
```
[the remark this answers is not in the document]
Comment (psorinionut) [reply auto-pos-creator/Cargo.toml:38]: Yes, for testing purposes until backTransfers are available. Will revert when we will have the new functionality reimplemented.

File: auto-pos-creator/src/lib.rs [hunk 3]
```diff
@@ -26,12 +23,11 @@ pub trait AutoPosCreator:
     + multi_contract_interactions::exit_pos_endpoints::ExitPosEndpointsModule
 {
     #[init]
-    fn init(&self, egld_wrapper_address: ManagedAddress, wegld_token_id: TokenIdentifier) {
+    fn init(&self, egld_wrapper_address: ManagedAddress, router_address: ManagedAddress) {
         self.require_sc_address(&egld_wrapper_address);
-        self.require_valid_token_id(&wegld_token_id);
-
+        self.require_sc_address(&router_address);
         self.egld_wrapper_sc_address().set(egld_wrapper_address);
```
[the remark this answers is not in the document]
Comment (psorinionut) [reply auto-pos-creator/src/lib.rs:29]: Updated.

File: auto-pos-creator/src/multi_contract_interactions/create_pos.rs (hunk 1 above)
Comment (psorinionut) [reply auto-pos-creator/src/multi_contract_interactions/create_pos.rs:31]: Removed.

File: auto-pos-creator/src/external_sc_interactions/egld_wrapper_actions.rs [hunk 4]
```diff
@@ -21,9 +21,6 @@ pub trait EgldWrapperActionsModule {
             .execute_on_dest_context()
     }
 
-    #[storage_mapper("wegldTokenId")]
-    fn wegld_token_id(&self) -> SingleValueMapper<TokenIdentifier>;
-
     #[storage_mapper("egldWrapperScAddress")]
```
[the remark this answers is not in the document]
Comment (psorinionut) [reply auto-pos-creator/src/external_sc_interactions/egld_wrapper_actions.rs:24]: Updated. Needed a new reference towards mx-contracts-rs repo in Cargo.toml.

**Review by CostinCarabas — APPROVED** [review]

**Review by alyn509 — APPROVED** [review]

**Pull request merged**